src/robot/Robot.cpp

```diff
@@ -1,21 +1,454 @@
+    }
+
+    auto spline
+        = dynamic_cast<const aikido::trajectory::Spline*>(rawPlan.get());
+    if (spline)
+    {
+      return postprocessor->postprocess(
+          *spline, *(cloneRNG().get()), testableConstraint);
+    }
+
+    // Else return raw path
+  }
+  return rawPlan;
+}
+
+//=============================================================================
+trajectory::TrajectoryPtr Robot::planToTSR(
+    const std::string bodyNodeName,
+    const constraint::dart::TSRPtr& tsr,
+    const constraint::TestablePtr& testableConstraint,
+    std::size_t maxSamples,
+    const std::shared_ptr<planner::ConfigurationToConfigurationPlanner>&
+        planner,
+    const std::shared_ptr<aikido::planner::TrajectoryPostProcessor>
+        trajPostProcessor,
+    const distance::ConstConfigurationRankerPtr& ranker) const
+{
+  // Get Body Node
+  auto bn = mMetaSkeleton->getBodyNode(bodyNodeName);
+  if (!bn)
+  {
+    dtwarn << "Request body node not present in robot '" << mName << "'"
+           << std::endl;
+    return nullptr;
+  }
+
+  // Create the problem.
+  auto problem = planner::dart::ConfigurationToTSR(
+      mStateSpace,
+      mStateSpace->getScopedStateFromMetaSkeleton(mMetaSkeleton.get()),
+      bn,
+      maxSamples,
+      tsr,
+      testableConstraint);
+
+  // Default to base Snap Planner
+  auto basePlanner = planner;
+  if (!basePlanner)
+  {
+    using planner::SnapConfigurationToConfigurationPlanner;
+    using statespace::GeodesicInterpolator;
+    basePlanner = std::make_shared<SnapConfigurationToConfigurationPlanner>(
+        mStateSpace, std::make_shared<GeodesicInterpolator>(mStateSpace));
+  }
+
+  // Convert to DART TSR planner.
+  auto tsrPlanner = std::make_shared<
+      planner::dart::ConfigurationToConfiguration_to_ConfigurationToTSR>(
+      basePlanner, mMetaSkeleton, ranker);
+
+  // Solve the problem with the DART planner.
+  auto rawPlan = tsrPlanner->plan(problem, /* result */ nullptr);
+
+  // Postprocess if enabled or provided
+  auto postprocessor
+      = (trajPostProcessor)
+            ? trajPostProcessor
+            : ((mEnablePostProcessing) ? mDefaultPostProcessor : nullptr);
+  if (rawPlan && postprocessor)
+  {
+    // Cast to interpolated or spline:
+    auto interpolated
+        = dynamic_cast<const aikido::trajectory::Interpolated*>(rawPlan.get());
+    if (interpolated)
+    {
+      return postprocessor->postprocess(
+          *interpolated, *(cloneRNG().get()), testableConstraint);
+    }
+
+    auto spline
+        = dynamic_cast<const aikido::trajectory::Spline*>(rawPlan.get());
+    if (spline)
+    {
+      return postprocessor->postprocess(
+          *spline, *(cloneRNG().get()), testableConstraint);
+    }
+
+    // Else return raw path
+  }
+  return rawPlan;
 }
 
 } // namespace robot
 } // namespace aikido
+
+// TODO: Switch to PRIMITIVE once this is fixed in DART.
+// mCollisionDetector->setPrimitiveShapeType(FCLCollisionDetector::PRIMITIVE);
```

Comment:
? This TODO and commented line are unclear to me. If it's important to keep, perhaps add some more context so that future generations of developers can pick up on this todo?